[docs] Adds monitoring doc (#32281)
expo/expo: docs/pages/monitoring/monitor-your-app.mdx

---
title: Monitor your app

> description: Learn how to monitor the usage of your app after its release.

Reviewer: ```suggestion
description: Learn how to monitor the usage of your Expo and React Native app after its release.
```

Let's update this to include the keywords for SEO.

---

import { BookOpen02Icon } from '@expo/styleguide-icons/outline/BookOpen02Icon';

import { BoxLink } from '~/ui/components/BoxLink';
import { ContentSpotlight } from '~/ui/components/ContentSpotlight';

Once your app is released, you can track anonymized usage data to give you insights on how users use your app. This data includes which updates are in use, when users experience bugs, and more.

## EAS Insights

Expo provides the [`expo-insights`](/eas-insights/introduction/) library, which tracks information related to [EAS Update](/deploy/send-over-the-air-updates/). This data includes the app version, platform, OS version, and update adoption. After you install it and release production builds on the app stores, you'll be able to see additional data on your project dashboard:

<ContentSpotlight
  alt="Active user insights on the Expo dashboard."
  src="/static/images/monitoring/monitor-your-app/expo-insights.png"
/>

Get started with the following guide:

<BoxLink
  title="EAS Insights"
  description="Learn how to use EAS Insights to monitor your app."
  href="/eas-insights/introduction/"
  Icon={BookOpen02Icon}
/>

## LogRocket

> You can get more insights with [LogRocket](https://logrocket.com). LogRocket records user sessions and identifies bugs as your users use your app. You can filter sessions by update IDs and also connect to your LogRocket account on the Expo dashboard to get quick acccess to your app's session data.

Reviewer: ```suggestion
You can get more insights with [LogRocket](https://logrocket.com). LogRocket records user sessions and identifies bugs as your users use your app. You can filter sessions by update IDs and also connect to your LogRocket account on the Expo dashboard to get quick access to your app's session data.
```

Typo.

<ContentSpotlight
  alt="User sessions on the LogRocket dashboard."
  src="/static/images/monitoring/monitor-your-app/logrocket.png"
/>

Get started with the following guide:

<BoxLink
  title="Using LogRocket"
  description="Learn how to use LogRocket to monitor your app."
  href="/guides/using-logrocket/"
  Icon={BookOpen02Icon}
/>

## Sentry

> [Sentry](http://getsentry.com/) is a crash reporting platform that provides you with real-time insight into production deployments with info to reproduce and fix crashes.

Reviewer: ```suggestion
[Sentry](http://getsentry.com/) is a crash reporting platform that provides real-time insight into production deployments with information to reproduce and fix crashes.
```

> It notifies you of exceptions or errors that your users run into while using your app and organizes them for you on a web dashboard. Reported exceptions include stacktraces, device info, version, and other relevant context automatically. You can also provide additional context that is specific to your application such as the current route and user id.

Reviewer: ```suggestion
It notifies you of exceptions or errors that your users run into while using your app and organizes them for you on a web dashboard. Reported exceptions include stacktraces, device info, version, and other relevant context automatically. You can also provide additional context that is specific to your app, such as the current route and user ID.
```

---
title: Monitor your app
description: Learn how to monitor the usage of your app after its release.
---

import { ContentSpotlight } from '~/ui/components/ContentSpotlight';
import { Terminal } from '~/ui/components/Snippet';

Once your app is released, you can track anonymized usage data to give you insights on how users use your app. This data includes which updates are in use, when users experience bugs, and more.

> ## Expo insights

Reviewer 2: ```suggestion
## Expo Insights
```

Expo provides the [`expo-insights`](/eas-insights/introduction/) library, which tracks information related to [EAS Update](/deploy/send-over-the-air-updates/). This data includes the app version, platform, OS version, and update adoption.

You can install `expo-insights` with:

> <Terminal cmd={['npx expo install expo-insights']} />

Reviewer: ```suggestion
<Terminal cmd={['$ npx expo install expo-insights']} />
```

Missing `$` prefix in all commands on the page. Let's add it so if there's any syntax highlights applicable, this will cover that.

After you install it and release production builds on the app stores, you'll be able to see additional data on your project dashboard:

<ContentSpotlight
  alt="Active user insights on the Expo dashboard."
  src="/static/images/monitoring/monitor-your-app/expo-insights.png"
/>

You can learn more about insights with our [EAS Insights guide](/eas-insights/introduction/).

## LogRocket

You can get more insights with [LogRocket](https://logrocket.com). LogRocket records user sessions and identifies bugs as your users use your app. To get started, you can install the LogRocket SDK with the following command:

> <Terminal cmd={['npx expo install @logrocket/react-native']} />

Reviewer: ```suggestion
<Terminal cmd={['$ npx expo install @logrocket/react-native']} />
```

Reviewer: Looking at the next snippet it seems setting `minSdkVersion` might be necessary in some cases. we should cover that (maybe a callout is enough to do so)?

Then, in your **app.json**/**app.config.ts**, include the LogRocket config plugin:

> ```json app.json/app.config.ts

Reviewer: ````suggestion
```json app.json
````

The syntax below is in contrast to `app.json,`, and we have already mentioned other variants. so if someone copy/pastes, they are aware of what they are doing and to avoid any confusion, let's remove the TypScript variant.

{
  "plugins": [
    [
      "expo-build-properties",
      {
        "android": {
          "minSdkVersion": 25
        }
      }
    ],
    "@logrocket/react-native"
  ]
}
```

> Finally, initialize LogRocket in your app in a top-level file, like **app/\_layout.ts** or **App.js**:

Reviewer: ```suggestion
Finally, initialize LogRocket in your app in a top-level file, like **app/\_layout.tsx** or **App.js**:
```

> ```ts app/_layout.ts/App.js

Reviewer: ````suggestion
```tsx app/_layout.tsx
````

As per default template, it is a tsx file. Also, let's remove `App.js` since we've already mentioned it in the text above. Reading this right now seems like there's either nesting of files going on it creates confusion.

> import React, { useEffect } from 'react';

Reviewer: ```suggestion
import { useEffect } from 'react';
```

I don't think importing `React` here is necessary.
